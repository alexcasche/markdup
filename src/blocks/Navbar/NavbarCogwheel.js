import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { tint } from 'polished'

const MODIFIERS = {
  open: ({ theme }) => `
    transform: rotate(45deg);
    color: ${tint(0.25, theme.colors.grey)};
  `,
  hover: ({ theme }) => `
    color: ${tint(0.25, theme.colors.grey)};
  `,
}

const NavbarCogwheel = styled(FontAwesomeIcon)`
  color: ${props => props.theme.colors.grey};
  font-size: 20px;
  transition: inherit;
  transform-origin: 50%;
  ${applyStyleModifiers(MODIFIERS)};
`

export default NavbarCogwheel
