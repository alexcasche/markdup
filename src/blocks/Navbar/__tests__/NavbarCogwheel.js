import { faCog } from '@fortawesome/free-solid-svg-icons'
import NavbarCogwheel from '../NavbarCogwheel'

describe('NavbarCogwheel Element', () => {
  it('renders correctly', () => {
    const tree = render(<NavbarCogwheel icon={faCog} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds open modifier', () => {
    const tree = render(
      <NavbarCogwheel modifiers={['open']} icon={faCog} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('transform', 'rotate(45deg)')
  })
})
