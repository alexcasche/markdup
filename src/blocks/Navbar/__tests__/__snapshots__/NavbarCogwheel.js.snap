// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarCogwheel Element adds open modifier 1`] = `
.c0 {
  color: #ABADB3;
  font-size: 20px;
  -webkit-transition: inherit;
  transition: inherit;
  -webkit-transform-origin: 50%;
  -ms-transform-origin: 50%;
  transform-origin: 50%;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  color: #eaeaec;
}

<svg
  aria-hidden="true"
  className="svg-inline--fa fa-cog fa-w-16 c0"
  data-icon="cog"
  data-prefix="fas"
  modifiers={
    Array [
      "open",
    ]
  }
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`NavbarCogwheel Element renders correctly 1`] = `
.c0 {
  color: #ABADB3;
  font-size: 20px;
  -webkit-transition: inherit;
  transition: inherit;
  -webkit-transform-origin: 50%;
  -ms-transform-origin: 50%;
  transform-origin: 50%;
}

<svg
  aria-hidden="true"
  className="svg-inline--fa fa-cog fa-w-16 c0"
  data-icon="cog"
  data-prefix="fas"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
